pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -59,66 +60,138 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 	// FP validation
 	fpDynamic, err := dynamic.NewValidator(dv.log, dv.env, dv.env.Environment(), dv.subscriptionDoc.ID, dv.fpAuthorizer, dynamic.AuthorizerFirstParty)
 	if err != nil {
-		return err
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidParameter,
+			"First Principal",
+			"Failed to validate the first principal",
+		)
 	}
 
 	err = fpDynamic.ValidateVnet(ctx, dv.oc.Location, subnets, dv.oc.Properties.NetworkProfile.PodCIDR, dv.oc.Properties.NetworkProfile.ServiceCIDR)
 	if err != nil {
-		return err
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidLinkedVNet,
+			"Cluster Virtual Network",
+			"Invalid virtual network specified",
+		)
```

Comment:
This will turn many specific errors into a very generic "Invalid virtual network specified" error. For example we will be hiding this:

https://github.com/Azure/ARO-RP/blob/4a48003b3e2345fda51ac3e860df4134cb494158/pkg/api/validate/dynamic/dynamic.go#L169

and this:

https://github.com/Azure/ARO-RP/blob/4a48003b3e2345fda51ac3e860df4134cb494158/pkg/api/validate/dynamic/dynamic.go#L173

And many others. I think this applies to the rest of the changes in the PR.

Reply:
Thanks for catching this oversight. In hindsight, I can't believe I missed it. I have added checks to ensure that existing CloudErrors are not inadvertently replaced by the "top-level" CloudError messages, since they are only meant to replace the generic `Internal Server Error` messages we send for vanilla `error` objects.